airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -26,18 +34,100 @@ def delete_dag_run():
     raise NotImplementedError("Not implemented yet.")
 
 
-def get_dag_run():
+@provide_session
+def get_dag_run(dag_id, dag_run_id, session):
     """
     Get a DAG Run.
     """
-    raise NotImplementedError("Not implemented yet.")
+    query = session.query(DagRun)
+    query = query.filter(DagRun.dag_id == dag_id)
+    query = query.filter(DagRun.run_id == dag_run_id)
+    dag_run = query.one_or_none()
+    if dag_run is None:
+        raise NotFound("DAGRun not found")
+    return dagrun_schema.dump(dag_run)
 
 
-def get_dag_runs():
+@provide_session
+def get_dag_runs(dag_id, session):
     """
     Get all DAG Runs.
     """
-    raise NotImplementedError("Not implemented yet.")
+    offset = request.args.get(parameters.page_offset, 0)
+    limit = min(int(request.args.get(parameters.page_limit, 100)), 100)
+
+    start_date_gte = parse_datetime_in_query(
```

Comment:
What do you think about [webargs](https://webargs.readthedocs.io/en/latest/) library? Would it be helpful to you?

Reply:
I found that this can solve it for us without any validation code from our side:

app.add_api("openapi.yaml", strict_validation=True)

https://connexion.readthedocs.io/en/latest/request.html#parameter-validation

Should I add it?

Reply:
I will be using webargs. Connexion no longer validate date-format because of license of one of its library.
https://github.com/zalando/connexion/issues/476

Reply:
I'm glad you did research. We can change it in a separate change. Now it works, so it's great. In next changes, we will be able to make a refactor and remove repetitive code fragments.